src/meta/src/rpc/ddl_controller.rs

```diff
@@ -758,7 +1022,18 @@ impl DdlController {
             .mark_creating_tables(&creating_tables)
             .await;
 
-        Ok((ctx, table_fragments))
+        // 5. If we have other tables that will be affected (for example, Sink into table),
+        // we need to modify them to generate a replace table plan to change their operational status.
+        let replace_table_job = if let Some(replace_table_info) = affected_table_replace_info {
+            Some(
+                self.inject_replace_table_job(env, &table_fragments, replace_table_info)
```

Comment:
The struct `replace_table_info` was passed via a deep code path. I am thinking that is it possible to simplify it like this?

```
RPC: create_sink -> run_command -> create_streaming_job -> build_stream_job -> inject_replace_table_job
⬇️
RPC: create_sink -> run_command (1) original code path
                 -> run_command (2) inject_replace_table_job
```

(or can be even better if they can be seperated into multiple RPCs but with atomicity... but this seems to be more difficult)